bin/recovery.sunxi-fel

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2016 Konsulko Group. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+# The sunxi-fel tool supports passing a bus:dev to use.  However this value
+# is not stable and may change over resets.  So at this time we assume that only
+# one board will be in FEL mode at a given time (or baring that, the first
+# available board is the one we are to test
+
+sunxi-fel uboot "${U_BOOT_BUILD_DIR}"/u-boot-sunxi-with-spl.bin
```

Comment:
I assume this downloads the U-Boot binary over USB/... If so, this file should be name download.sunxi-fel not recovery.sunxi-fel.

The naming convention is:

recovery.xxx gets the board into recovery mode; a mode in which you can download a binary to the board.
download.xxx communicates with a board that's in recovery mode, and actually sends the binary.

In reset.tegra-usb-recovery, you'll see a call to recovery.* which calls a utility to manipulate reset and recovery-mode buttons/signals, followed by a call to download.* to send the binary using e.g. tegra-uboot-flasher.